packages/runtime-core/src/renderer.ts

```diff
@@ -1328,6 +1328,10 @@ function baseCreateRenderer(
               parentSuspense,
               null,
             )
+            // when data-allow-mismatch is set, the component vnode's el may be incorrect, so it should be updated
+            if (initialVNode) {
```

Comment:
The `vnode.el` needs to be updated only if a mismatch occurs.
So this is not the proper fix. The correct fix should be in the `handleMismatch` function.
- update `vnode.el` as follows: https://github.com/vuejs/core/blob/ed01d9257104cf59d8330d96af77941b40494aa1/packages/runtime-core/src/components/Suspense.ts#L896-L897